doc/internal/man3/DEFINE_SPARSE_ARRAY_OF.pod

```diff
@@ -0,0 +1,109 @@
+=pod
+
+=head1 NAME
+
+DEFINE_SPARSE_ARRAY_OF, sa_TYPE_new, sa_TYPE_free, sa_TYPE_free_leaves,
+sa_TYPE_num, sa_TYPE_doall, sa_TYPE_doall_arg, sa_TYPE_get, sa_TYPE_set
+- sparse array container
+
+=head1 SYNOPSIS
+
+ #include "internal/sparse_array.h"
+
+ typedef struct sparse_array_st OPENSSL_SA;
+
+ SPARSE_ARRAY_OF(TYPE)
+ DEFINE_SPARSE_ARRAY_OF(TYPE)
+
+ SPARSE_ARRAY_OF(TYPE) *sa_TYPE_new(void);
+ void sa_TYPE_free(const SPARSE_ARRAY_OF(TYPE) *sa);
+ void sa_TYPE_free_leaves(const SPARSE_ARRAY_OF(TYPE) *sa);
+ int sa_TYPE_num(const SPARSE_ARRAY_OF(TYPE) *sa);
+ void sa_TYPE_doall(const OPENSSL_SA *sa, void (*leaf)(void *));
+ void sa_TYPE_doall_arg(const OPENSSL_SA *sa, void (*leaf)(void *), void *arg);
+ TYPE *sa_TYPE_get(const SPARSE_ARRAY_OF(TYPE) *sa, size_t idx);
+ int sa_TYPE_set(SPARSE_ARRAY_OF(TYPE) *sa, size_t idx, TYPE *value);
```

Comment:
I have to admit, I am apprehensive of the `sa_` prefix, and am wonder if `ossl_sa_` wouldn't be more suitable, even though this is internal stuff.

I know that this is annoying (believe me), but we have a problem with a proliferation of all kinds of prefixes, and even though this is internal stuff, you never know if this will clash with some public API that some .c file just happens to include at the same time as using this.

Reply:
Is _sa_ any worse than _lh_ or _sk_ which are public?

I'm willing to change it.

Reply:
lh and sk are public for historical reasons and are a mistake not likely to be repeated.
all new functions should be prefixed.

Reply:
Changed.  Unless the build and tests fail :)

Reply:
We made an executive decision to use `ossl_` or `openssl_` as a prefix for new *APIs* (so for a new API `foo`, the API would really be named `ossl_foo` or `openssl_foo`) around the time I worked on STORE (which got renamed to `OSSL_STORE`).

We're a bit more lax with already existing prefix (see `EVP_MAC`, since `EVP_` is an already existing prefix).